opm/parser/eclipse/EclipseState/Schedule/Well.cpp

```diff
@@ -303,11 +526,109 @@ namespace Opm {
         }
         return wellNameInPattern;
     }
+    bool Well::isMultiSegment() const {
+        return m_is_multi_segment;
+    }
+
+    void Well::setMultiSegment(const bool is_multi_segment) {
+        m_is_multi_segment = is_multi_segment;
+    }
+
+    SegmentSetConstPtr Well::getSegmentSet(size_t time_step) const {
+        return m_segmentset->get(time_step);
+    }
+
 
     bool Well::getAllowCrossFlow() const {
         return m_allowCrossFlow;
     }
 
+    void Well::processCOMPSEGS(size_t time_step, std::vector<CompsegsPtr>& compsegs) {
+
+        // TODO: make sure the distance information of compsegs is complete.
+        // TODO: for the current cases we have at the moment, the distance information is specified explicitly,
+        // TODO: while the depth information is defaulted though
+        // process the compsegs to get the segment_information
+        SegmentSetPtr currentSegmentSet = m_segmentset->get(time_step);
+
+        for (size_t i_comp = 0; i_comp < compsegs.size(); ++i_comp) {
+            if (compsegs[i_comp]->m_segment_number == 0) { // need to determine the segment number first
+                const double center_distance = (compsegs[i_comp]->m_distance_start + compsegs[i_comp]->m_distance_end) / 2.0;
+                const int branch_number = compsegs[i_comp]->m_branch_number;
+
+                int segment_number = 0;
+                double min_distance_difference = 1.e100; // begin with a big value
+                for (int i_segment = 0; i_segment < currentSegmentSet->numberSegment(); ++i_segment) {
```

Comment:
fixed. I was wondering how to remove the warnings about comparison between int and size_t.